cpp/include/cugraph/graph_view.hpp

```diff
@@ -958,6 +979,18 @@ class graph_view_t<vertex_t,
     return vertex_partition_view_t<vertex_t, false>(this->number_of_vertices());
   }
 
+  void attach_mask_view(graph_mask_view_t<vertex_t, edge_t>& mask)
+  {
+    mask_ = std::make_optional<graph_mask_view_t<vertex_t, edge_t>*>(&mask);
+  }
+
+  std::optional<graph_mask_view_t<vertex_t, edge_t>> get_mask_view(size_t partition_idx = 0) const
+  {
+    assert(partition_idx == 0);
+    return mask_.has_value() ? std::make_optional<graph_mask_view_t<vertex_t, edge_t>>(**mask_)
+                             : std::nullopt;
+  }
+
```

Comment:
And we should have something like `clear_mask_view` as well setting `mask_` to std::nullopt;